jsonrpc/eth_endpoint.go

```diff
@@ -760,19 +733,9 @@ func (e *Eth) GetTransactionCount(address types.Address, filter BlockNumberOrHas
 
 // GetCode returns account code at given block number
 func (e *Eth) GetCode(address types.Address, filter BlockNumberOrHash) (interface{}, error) {
-	var (
-		header *types.Header
-		err    error
-	)
-
-	// The filter is empty, use the latest block by default
-	if filter.BlockNumber == nil && filter.BlockHash == nil {
-		filter.BlockNumber, _ = createBlockNumberPointer("latest")
-	}
-
-	header, err = e.getHeaderFromBlockNumberOrHash(&filter)
+	header, err := e.getHeaderFromBlockNumberOrHash(filter)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get header from block hash or block number")
+		return nil, err
```

Comment:
Shall we add block number or hash to the error?

Reply:
The error from `getHeaderFromBlockNumberOrHash` already includes the context (num or hash error and the value).